lib/nodejs/supervisor.ex

```diff
@@ -29,38 +29,32 @@ defmodule NodeJS.Supervisor do
     Supervisor.stop(__MODULE__)
   end
 
-  defp to_transaction(module, args, opts) do
+  defp run_in_transaction(module, args, opts) do
     timeout = Keyword.get(opts, :timeout, @timeout)
     binary = Keyword.get(opts, :binary, false)
 
     func = fn pid ->
       try do
-        GenServer.call(pid, {module, args, binary}, timeout)
+        GenServer.call(pid, {module, args, binary}, :infinity)
```

Comment:
Wondering if it is safe to set this to `:infinity`.

Reply:
I guess not. I checked the source code for poolboy, and the timeout of `:poolboy.transaction` only applies to the checkout call, the function itself will be blocking.
It's probably best to just use the same timeout again. However, this would mean that it could theoretically take `2*timeout` until a timeout happens, since there will be `:poolboy.checkout` time + `GenServer.call` time.
But I think that's ok, since as a caller I usually don't care how long exactly the timeout takes, just that it does happen within reasonable time.

I'll give this another look and will update the PR.

Reply:
Is there a test that already checks for this? If not I'll see if I can write a good test for this. E.g. make sure the workers aren't blocked when JS functions never return.

Reply:
It seems that the test to override timeout should handle it, but we may want to double check them with these changes.

Reply:
This is a bit more involved than I hoped. I added a new failing test. The new test checks if we can still use `NodeJS.call` after a slow running js function got a timeout.
Both the original implementation and my new changes fail this test.
I'll dig a bit deeper. It looks like we also need a timeout in the workers

Reply:
Ok, I think I found a decent solution now. Would be nice if you could check again :smiley: